core-rust/core-api-server/src/core_api/handlers/mempool_transaction.rs

```diff
@@ -6,19 +6,55 @@ pub(crate) async fn handle_mempool_transaction(
 ) -> Result<Json<models::MempoolTransactionResponse>, ResponseError<()>> {
     assert_matching_network(&request.network, &state.network)?;
 
+    if request.payload_hashes.len() > MAX_BATCH_COUNT_PER_REQUEST.into() {
+        return Err(client_error(format!(
+            "you can query up to {MAX_BATCH_COUNT_PER_REQUEST} payload hashes"
+        )));
+    }
+
+    let mut response = models::MempoolTransactionResponse {
+        count: MAX_BATCH_COUNT_PER_REQUEST as i32, // placeholder to get a better size aproximation for the header
+        payloads: vec![],
+    };
+
     let extraction_context = ExtractionContext::new(&state.network);
-    let payload_hash =
-        extract_notarized_transaction_hash(&extraction_context, request.payload_hash)
-            .map_err(|err| err.into_response_error("payload_hash"))?;
+
+    let mut current_total_size = response.get_json_size();
```

Comment:
AFAIS, we now have 4 places where this "output stuff until size limit reached" is implemented. Would it be possible to handle it in a piece of infra? (ofc can be in a post-refactor PR)